### ISSUE TITLE
Fix random seeds in model tests

### ISSUE THREAD

**Issue by MischaPanch** [jambit/sensAI#32]
Sometimes (rarely) they fail due to desired accuracy not being reached

**Comment by opcode81** [comment]
Please be more specific. Which test(s) failed?

**Comment by MischaPanch** [comment]
> Please be more specific. Which test(s) failed?

Unfortunately, I had restarted the test (it was in CI) so I don't have the log anymore. I don't remember which one it was. When the error reappears, we can link the log to this issue.

**Comment by opcode81** [comment]
Here's an example of a test that failed randomly (sklearn's MLP with LBFGS):
https://github.com/jambit/sensAI/runs/2876706830?check_suite_focus=true#step:8:137

**Comment by MischaPanch** [comment]
Although here another bug made its appearance (before reaching the accuracy comparison): the warning message is not raised properly, result.message seems to be a string and does not have .decode (line 145 in the log)

```
        if solver == "lbfgs":
            if result.status != 0:
                warning_msg = (
                    "{} failed to converge (status={}):\n{}.\n\n"
                    "Increase the number of iterations (max_iter) "
                    "or scale the data as shown in:\n"
                    "    https://scikit-learn.org/stable/modules/"
                    "preprocessing.html"
>               ).format(solver, result.status, result.message.decode("latin1"))
E               AttributeError: 'str' object has no attribute 'decode'
```

**Comment by opcode81** [comment]
Yes, that's in sklearn though, not our code.

**Comment by opcode81** [comment]
Fixed the issue that was encountered in 0ce6878f2c7c6f12a4387fa2089b2e73ca96d80f
